proposals/lip-0053.md

```diff
@@ -438,26 +679,20 @@ TBA
 
 ### Computing the Validators Update
 
-When posting a CCU transaction, the validators hash given in the certificate certifies the new set of validators of the sending chain. The CCU must therefore include the difference between the validators currently stored in the chain account and the validator set authenticated by the certificate. The required `activeValidatorsUpdate` can be obtained by applying the function below. We suppose that `currentValidators` and `newValidators` follow the `activeValidators` schema given above.
+When posting a CCU transaction, the validators hash given in the certificate certifies the new set of validators of the sending chain. The CCU must therefore include the difference between the validators currently stored in the chain account and the validator set authenticated by the certificate. The required `activeValidatorsUpdate` can be obtained by applying the function below.
 
 ```python
-getActiveValidatorsDiff(currentValidators, newValidators):
+def getActiveValidatorsDiff(currentValidators: list[ActiveValidator], newValidators: list[ActiveValidator]) -> list[ActiveValidator]:
     activeValidatorsUpdate = []
 
-    for newValidator in newValidators.activeValidators:
-        if there exist currentValidator in currentValidators.activeValdators such that
-        newValidator.blsKey == currentValidator.blsKey
-	and newValidator.bftWeight == currentValidator.bftWeight:
-            continue
-        else:
-            append newValidator to activeValidatorsUpdate
+    for validator in newValidators:
+        if validator not in currentValidators:
+            activeValidatorsUpdate.append(validator)
 
-    for currentValidator in currentValidators:
-        if there exist newValidators in currentValidators.activeValdators such that
-        newValidator.blsKey == currentValidator.blsKey:
-            continue
-        else:
-            append {"blsKey": currentValidator.blsKey, "bftWeight": 0} to activeValidatorsUpdate
+    currentKeys = [validator.blsKey for validator in currentValidators]
+    for validator in currentValidators:
+        if validator.blsKey not in currentKeys:
```

Comment:
This condition will never be `True` as you only consider the current validators and current BLS keys.

Reply:
Indeed, I had a mistake in the pseudocode. fixed